### DISCUSSION TITLE
How to load a raw timestream from a session ID

### DISCUSSION THREAD

**Discussion by samdayweiss** [simonsobs/sodetlib#360] [Q&A] [answered]
Hi All!

 For a novice on sotodolib: Does anyone have a code example of how to access a raw timestream (including converting to pA) from a session ID?

Thanks!

**Comment by jlashner** [comment, marked as answer]
Hi sam! If you have the stream-id and the session-id, you can use the `load_session` function in `sodetlib.stream` to load an axis manager. This is loaded into the base `sodetlib` package, so you can do something like:
```
import sodetlib as sdl

stream_id = <stream_id>
session_id = <session_id>
am = sdl.load_session(stream_id, session_id)
```
If the data has already been transferred and you are running this on a server that's not the smurf-server, you need to set the base directory where the timestreams are saved, and pass that in as the `base_dir` kwarg. For example:
```
am = sdl.load_session(stream_id, session_id, base_dir='/mnt/so1/data/ucsd-sat1/timestreams')
```

If you have a G3tSmurf database setup, there are other ways that we use to interact with archival data, which are [documented here](https://sotodlib.readthedocs.io/en/latest/g3tsmurf.html). Hope this helps!

**Reply by samdayweiss** [reply]
If the data is from noise.take_noise(), it seems to only have a session ID and no stream ID, will that function still work?

**Reply by jlashner** [reply]
If you have the result from `take_noise`, then the stream_id should be saved in `res['meta']['stream_id']`. (This is true for all sodetlib data products)
